pandas/_libs/groupby.pyx

```diff
@@ -1057,7 +1056,7 @@ def group_nth(rank_t[:, :] out,
 
             for i in range(ncounts):
                 for j in range(K):
-                    if nobs[i, j] == 0:
+                    if nobs[i, j] < min_count or nobs[i, j] == 0:
```

Comment:
Is this tested?

Reply:
Indirectly through cython tests, but is better to add this too. Adjusted the test to reflect the behavior